controllers/mysql_conf.go

```diff
@@ -0,0 +1,197 @@
+package controllers
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/cybozu-go/moco"
+	"github.com/go-logr/logr"
+)
+
+var (
+	confKeyPrefixes = []string{
+		"enable_",
+		"disable_",
+		"skip_",
+	}
+
+	// Default options of mysqld section
+	defaultMycnf = map[string]string{
+		"tmpdir":                              moco.TmpPath,
+		"innodb_tmpdir":                       moco.TmpPath,
+		"character_set_server":                "utf8mb4",
+		"collation_server":                    "utf8mb4_unicode_ci",
+		"skip_character_set_client_handshake": "ON",
+
+		"default_time_zone": "+0:00",
+
+		"back_log":            "900",
+		"max_connections":     "100000",
+		"max_connect_errors":  "10",
+		"max_allowed_packet":  "1G",
+		"max_heap_table_size": "64M",
+		"sort_buffer_size":    "4M",
+		"join_buffer_size":    "2M",
+		"thread_cache_size":   "100",
+		"wait_timeout":        "604800", // == 7 days
+		"lock_wait_timeout":   "60",
+
+		"table_open_cache":       "65536", // The default setting causes MY-012102
+		"table_definition_cache": "65536", // mitigate a innodb table cache eviction.
+
+		"transaction_isolation": "READ-COMMITTED",
+		"tmp_table_size":        "64M",
+		"slow_query_log":        "ON",
+		"long_query_time":       "2",
+		"log_error_verbosity":   "3",
+
+		"max_sp_recursion_depth": "20",
+
+		"print_identified_with_as_hex": "ON",
+
+		"loose_binlog_transaction_compression": "ON", // It would reduce the size of binlog by a third, but only available in 8.0.20 or later.
+
+		"information_schema_stats_expiry": "0", // No need to cache information_schema.tables values
+
+		// INNODB Specific options
+		"innodb_flush_method":              "O_DIRECT",
+		"innodb_lock_wait_timeout":         "60",
+		"innodb_print_all_deadlocks":       "1",
+		"innodb_online_alter_log_max_size": "1073741824",
+		"innodb_adaptive_hash_index":       "ON",
+		"innodb_numa_interleave":           "ON",
+
+		// options for adaptive flushing
+		"innodb_log_file_size":      "800M",
+		"innodb_log_files_in_group": "2",
+		"innodb_read_io_threads":    "8",
+		"innodb_write_io_threads":   "4",
+		"innodb_io_capacity":        "400",
+		"innodb_io_capacity_max":    "3200",
+		"innodb_flushing_avg_loops": "10",
+		"innodb_purge_threads":      "1",
+		"innodb_page_cleaners":      "1",
+		"innodb_lru_scan_depth":     "256",
```

Comment:
Are these optimized for SSDs?  It seems to me, for example, `innodb_io_capacity` is too small for SSDs.